.github/workflows/on_pr.yml

```diff
@@ -0,0 +1,49 @@
+name: lint and diff
+
+on:
+  workflow_call:
+
+jobs:
+  prettier_deploy_config:
+    runs-on: [self-hosted, ubuntu20.04-self]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actionsx/prettier@v2
```

Comment:
What's the use of this and lint.yml too?

Reply:
this is the workflow that will be include by other repos (and won't be used by this repo). 
it lints the ./deploy config for all repos. 

the lint.yml will lint this project. 

on_pr and on_push have to be in .github/workflows btw so no option to split the workflows